Llm connect backend

eyajel opened the pull request.

eyajel committed:
Save all new changes before cleaning up PR

eyajel committed:
Temporary backup work

eyajel committed:
Merge branch 'backup-branch' into issue-7-llm-connect-backend

eyajel committed:
LLM connect

tushartushar commented on coderefine-backend/app/db/repositories/suggestions_repo.py:
```diff
@@ -3,72 +3,123 @@
+    return (s or "").strip()
+
+
 def save_suggestions_structured(
```
Too long method.

tushartushar commented on coderefine-backend/app/llm/providers/llm_provider.py:
```diff
@@ -0,0 +1,255 @@
+        "syntactic_valid": is_valid_identifier_name(language.value, text),
+    }
+
+class LLMProvider:
```
split long methods into smaller methods

tushartushar commented on coderefine-backend/app/workers/tasks.py:
```diff
@@ -22,40 +24,72 @@ async def suggest_identifier_task(
+    provider_context: Dict[str, Any] | None = None,
+    syntactic_is_bad: Optional[bool] = None,
 ) -> Dict[str, Any]:
     """
```
too long method

tushartushar commented on coderefine-backend/app/workers/tasks.py:
```diff
@@ -22,40 +24,72 @@ async def suggest_identifier_task(
             language=language,
             code=code,
             identifier=name,
-            syntactic_is_bad=bool(verdict["is_bad"]),
```
looks like we are dealing with string literals ("is_bad") rather than boolean or enumerations

tushartushar commented on coderefine-backend/scripts/llm_ping_ollama.py:
```diff
@@ -0,0 +1,30 @@
+
+load_dotenv()
+
+BASE = os.getenv("CODEREFINE_LLM_API_BASE", "http://localhost:11434/v1")
```
is the readme mentions that we need to define all these env variables?

tushartushar commented on coderefine-backend/scripts/llm_simple_prompt_ollama.py:
```diff
@@ -0,0 +1,40 @@
+    payload = {
+        "model": MODEL,
+        "messages": [
+            {"role": "system", "content": "You are a helpful assistant."},
```
is the system prompt good enough? (it is not telling the llm what kind of helpful assistant?)

eyajel replied:
In that specific file , i'm just using a generic prompt to ping the model and confirm connectivity and if the model can actually respond ( just asking the model to say :" hello, i'm working "), The actual system prompt, is defined under the main pipeline in the llm provider where i'm using the prompt "You are a code renamaing expert"  .  You are absolutely right though , since i have other scripts that perform actual functional tests , this file isn't really relevant and not needed , i'll remove it to keep things clean !

tushartushar reviewed (changes requested):
please address the suggestions

eyajel committed:
Address PR comments , refactored code, added llm frontend wiring, added ollama as another suggestion engine along with cloud and llm , secured frontend and backend end to end flow

eyajel commented:
I’ve updated the code to address the PR comments, refactored parts, wired the frontend and backend for the LLM integration, added Ollama as another suggestion engine along with the local model and cloud LLMs, and secured the frontend and backend end-to-end flow.
